workflows/crisprseq_screening.nf

```diff
@@ -125,19 +139,57 @@ workflow CRISPRSEQ_SCREENING {
         )
         ch_versions = ch_versions.mix(FASTQC.out.versions.first())
 
-        ch_input_cutadapt = ch_input.combine(Channel.value([[]]))
+        //set adapter seq to null to make it compatible with crispr targeted
+        ch_cutadapt = ch_input.combine(Channel.value([[]]))
 
-    if(params.cutadapt) {
-        CUTADAPT(
-            ch_input_cutadapt
-        )
-        ch_versions = ch_versions.mix(CUTADAPT.out.versions)
+        if(params.five_prime_adapter) {
+            CUTADAPT_FIVE_PRIME(
+                ch_cutadapt
+            )
+            CUTADAPT_FIVE_PRIME.out.reads.combine(Channel.value([[]])).set { ch_cutadapt }
+            ch_cutadapt.map{ meta, fastq, proto  ->
+                meta.id = "${meta.id}_trim"
+                [meta, [fastq], proto]
+            }.set { ch_cutadapt }
 
-        CUTADAPT.out.reads
-        .map{ meta, fastq  ->
-            [meta, [fastq]]
+            ch_versions = ch_versions.mix(CUTADAPT_FIVE_PRIME.out.versions)
         }
-        .set { ch_input }
+
+        if(params.three_prime_adapter) {
+            CUTADAPT_THREE_PRIME(
+                ch_cutadapt
+            )
+            ch_cutadapt = CUTADAPT_THREE_PRIME.out.reads.combine(Channel.value([[]]))
+            ch_versions = ch_versions.mix(CUTADAPT_THREE_PRIME.out.versions)
+        }
+
+
+        if(params.five_prime_adapter || params.three_prime_adapter) {
+            ch_cutadapt
+            .map{ meta, fastq, empty  ->
+                [meta, [fastq]]
+            }
+            .set { ch_input }
+        }
+
+        if(params.fasta){
+            Channel.of("fasta")
+                .combine(Channel.fromPath(params.fasta))
+                .set{ ch_fasta }
+
+            BOWTIE2_BUILD(ch_fasta)
+            ch_versions = ch_versions.mix(BOWTIE2_BUILD.out.versions)
+
+            BOWTIE2_ALIGN (
+            ch_input,
+            BOWTIE2_BUILD.out.index,
+            false,
+            false
+            )
+
+            BOWTIE2_ALIGN.out.aligned.map{ meta, bam ->
+                [meta, [bam]]
+            }.set{ch_input}
```

Comment:
As mentioned before, I would prefer to avoid overriding the channel by using the same name.

One alternative could be use a placeholder, eg. `ch_input_preprocesses = ch_input` and `set{ch_input_preprocesses}` inside the bowtie `if`.